File: practice/java/DisplayPrimeNumbersBetweenTwoIntervals.java

import java.util.Scanner;

class DisplayprimeNumbersBetweenTwoIntervals {

    /**
     * @brief this function display prime numbers between two intervals
     * @param start_number: the starting number of the interval
     * @param end_number: the ending number of the interval
     * @return void: the function returns 1 if successful, 0 if an error occurs
     */
    static void prime_number_between_two_intervals(int start_number, int end_number) {
        for (int number = start_number; number <= end_number; number++) {
            int isprime = 1;
            if (number < 2) {
                isprime = 0;
            } else if (number != 2 && number != 3 && (number % 2 == 0 || number % 3 == 0)) {
                isprime = 0;
            }
            for (int i = 5; i * i < number; i += 2) {
                if (number % i == 0 || number % (i + 2) == 0) {
                    isprime = 0;
                }

                if (isprime == 1) {
                    System.out.printf("it a prime number:%d\n", number);
                }
            }
        }

    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter the starting number: ");
        int start_number = scanner.nextInt();
        System.out.print("Enter the ending number: ");
        int end_number = scanner.nextInt();
        scanner.close();
        prime_number_between_two_intervals(start_number, end_number);
    }

}

File: practice/java/display_prime_numbers_between_two_intervals.py
#@brief this function display prime numbers between two intervals
#@param start: starting interval
#@param end: ending interval
#@return: None
def display_prime_numbers_between_two_intervals(start_number, end_number):
    for number in range(start_number, end_number + 1):
        isprime = True
        if(number < 2):
            isprime = False
        if ( number !=2 and number != 3  and ( number % 2 == 0 or number % 3 == 0)):
            isprime = False
        for i in range(5, int(number ** 0.5) + 1, 6):
            if (number % i == 0 or number % (i + 2) == 0):
                isprime = False
                break
        if isprime:
            print("it is a prime number: ", number)


print("Enter the starting interval: ")
start_number = int(input())
print("Enter the ending interval: ")
end_number = int(input())
display_prime_numbers_between_two_intervals(start_number, end_number)